### PR TITLE
Skunkworks 2024!

### PR DESCRIPTION
**Pull request by vstehlik**
- Updates to dependencies

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -182,19 +189,21 @@ export const createUser: Handler = async (event: any) => {
...
+  const tokenResult = await oauthClient.getToken({
     code,
-    redirect_uri: createUserUrl(),
+    redirect_uri: createUserUrl()
   });
-  const accessToken = oauthClient.accessToken.create(tokenResult);
-  const tokenStr: string = accessToken.token.access_token;
+  const tokenStr: string = tokenResult.token.access_token as string;
+  const user: string = tokenResult.token.user_id as string;
+  console.log(tokenResult.token);
```
Comment (vstehlik) [inline src/index.ts:201]: Remove

File: src/index.ts [hunk 2]
```diff
@@ -182,19 +189,21 @@ export const createUser: Handler = async (event: any) => {
...
 
-  console.log(`AccessToken=${JSON.stringify(accessToken)}, TokenStr=${tokenStr}`);
+  const slackClient = new WebClient(tokenStr);
+  const userInfo = await slackClient.users.info({ token: tokenStr, user: user })
 
-  const slackClient = new WebClient(tokenStr, {logLevel: LogLevel.DEBUG});
-  const userInfo = await slackClient.users.info();
+  console.log(userInfo);
```
Comment (vstehlik) [inline src/index.ts:206]: Remove

File: src/services/calendar/graphApiAuthenticationProvider.ts [hunk 3]
```diff
@@ -71,26 +72,25 @@ export class GraphApiAuthenticationProvider implements AuthenticationProvider {
     }
 
     console.log(
-      `Microsoft Graph access token expired for ${this.userEmail} at ${
-        this.storedToken.expires_at_timestamp
-      }. Refreshing...`,
+      `Microsoft Graph access token expired for ${this.userEmail}. Refreshing...`,
```
Comment (vstehlik) [inline src/services/calendar/graphApiAuthenticationProvider.ts:75]: Should we add timestamp back in to this message now that we are storing it in the db again?

File: src/index.ts (hunk 1 above)
Comment (ppoulsen) [reply src/index.ts:201]:
```suggestion
```

File: src/index.ts (hunk 2 above)
Comment (ppoulsen) [reply src/index.ts:205]:
```suggestion
```

File: src/services/calendar/graphApiAuthenticationProvider.ts (hunk 3 above)
Comment (ppoulsen) [reply src/services/calendar/graphApiAuthenticationProvider.ts:75]: Ooh, good call! Adding that now...

File: src/slackbot.ts [hunk 4]
```diff
@@ -261,10 +261,10 @@ You need to authorize me before we can do anything else: ${slackInstallUrl()}`);
   }
   const command = text;
   const tokens = command.match(/emoji\s*=\s*:[^"”:]+:|[\w-]+\s*=\s*(true|false|["“][^"”]+["”])|[^ "“”]+/gi) || [];
-  const subcommand = tokens[0];
+  const subcommand = tokens[0] || "none";
   const args = tokens.slice(1);
 
-  if (subcommand in commandHandlerMap) {
+  if (subcommand && subcommand in commandHandlerMap) {
```
Comment (vstehlik) [inline src/slackbot.ts:267]:
```suggestion
  if (subcommand in commandHandlerMap) {
```

File: src/services/slack.ts [hunk 5]
```diff
@@ -46,7 +44,8 @@ export const setUserPresence = async (email: string, token: string | undefined,
   const slackClient = new WebClient(token);
 
   try {
-    await slackClient.users.setPresence({ presence });
+    const result = await slackClient.users.setPresence({ presence });
+    throw result.error;
```
Comment (vstehlik) [inline src/services/slack.ts:48]: Do we want to throw the error or just log it?

**Review by ppoulsen — APPROVED** [review]

File: src/services/slack.ts (hunk 5 above)
Comment (ppoulsen) [reply src/services/slack.ts:48]: I think throwing is still good 🤔 The handleError function ends up checking the error message and if it's "token_revoked" or "invalid_auth"  it clears the tokens and sends the user a message in Slack to re-auth.

Comment (vstehlik) [reply src/services/slack.ts:48]: Does this need to be wrapped in a conditional statement then to only throw if there is an error?

Comment (ppoulsen) [reply src/services/slack.ts]: Swapped to the the response `ok` boolean and only throw if that's missing/false!
[2079346](https://github.com/hudl/CalendarToSlack/pull/96/commits/2079346de455257bba346d6c58f23420e848e340)

**Pull request merged**